docs/diagrams/PLUGINDIAGRAM.md

```diff
@@ -0,0 +1,100 @@
+# Automation LibOCR Plugin Flow Diagram
+
+```mermaid
+sequenceDiagram
+    participant LibOCR
+    participant Plugin
+    participant Observer
+    participant Coordinator
+    participant Encoder
+
+    par libOCR round sequence
+        LibOCR->>Plugin: Query
+        Note over LibOCR,Plugin: query is not required by automation
+        Plugin-->>LibOCR: [empty bytes]
+        LibOCR->>Plugin: Observation(Query)
+        Plugin->>Observer: Observe(Config)
+
+        Observer->>Observer: getLatestObserved
+
+        loop each upkeep
+            Observer->>Coordinator: IsPending()
+            Coordinator-->>Observer: bool
+            
+            Observer->>Observer: addToObservation
+
+            Observer->>Encoder: Eligible()
+            Encoder-->>Observer: bool
+        end
```

Comment:
Here is where the observer is constantly checking eligibility correct? The encoder is not really doing any work its just the translator for the observer as the observer doesnt want to interpret the UpkeepResult itself?

Reply:
Correct. This helps abstract the idea of a polling observer from any specific contract. All we need is a single observer type (polling, log trigger, cron, etc.) that can pair with any number of contracts provided the concept of eligible remains the same and an encoder is paired with the observer.